nuxt/plugin.js

```diff
@@ -1,61 +1,79 @@
-<% if (options.hasGlobalActions) { %>import * as globalActions from '~/<%= options.baseDir %>/actions'<% } %>
-<% if (options.hasGlobalGetters) { %>import * as globalGetters from '~/<%= options.baseDir %>/getters'<% } %>
+<% if (options.hasGlobalActions) { %>import * as actions from '~/<%= options.baseDir %>/actions'<% } %>
+<% if (options.hasGlobalGetters) { %>import * as getters from '~/<%= options.baseDir %>/getters'<% } %>
 <%
-const statorActionsModules = {}
-const statorGettersModules = {}
+const modules = {}
 for (const statorModule of options.statorModules) {
-  if (statorModule.actions) {
-    const importName = `statorActions_${statorModule.namespace}`
-    statorActionsModules[statorModule.namespace] = importName
+  const { namespace, state, getters, actions } = statorModule
+  if (!(state || getters || actions)) {
+    continue
+  }
+
+  modules[namespace] = {}
+
+  if (state) {
+    const importName = `${namespace}State`
+    modules[namespace].state = importName
 
-%>import * as <%= importName %> from '~/<%= options.baseDir %>/<%= statorModule.actions %>'<%
+%>import * as <%= importName %> from '~/<%= options.baseDir %>/<%= state %>'<%
   }
 
-  if (statorModule.getters) {
-    const importName = `statorGetters_${statorModule.namespace}`
-    statorGettersModules[statorModule.namespace] = importName
+  if (actions) {
+    const importName = `${namespace}Actions`
+    modules[namespace].actions = importName
 
-%>import * as <%= importName %> from '~/<%= options.baseDir %>/<%= statorModule.getters %>'<%
+%>import * as <%= importName %> from '~/<%= options.baseDir %>/<%= actions %>'<%
+  }
+
+  if (getters) {
+    const importName = `${namespace}Getters`
+    modules[namespace].getters = importName
+
+%>import * as <%= importName %> from '~/<%= options.baseDir %>/<%= getters %>'<%
   }
 } %>
 
 <% if (!options.hasState) { %>const state = () => ({})<% } %>
-<% if (!options.hasGlobalActions) { %>const globalActions = {}<% } %>
-<% if (!options.hasGlobalGetters) { %>const globalGetters = {}<% } %>
+<% if (!options.hasGlobalActions) { %>const actions = {}<% } %>
+<% if (!options.hasGlobalGetters) { %>const getters = {}<% } %>
 
-const actions = { <%= Object.entries(statorActionsModules).map(([k, v]) => `${k}: ${v}`).join(', ') %> }
-const getters = { <%= Object.entries(statorGettersModules).map(([k, v]) => `${k}: ${v}`).join(', ') %> }
+const modules = {
+  <%= Object.entries(modules).map(([moduleName, module]) =>
+  `${moduleName}: {
+    ${Object.entries(module).map(([key, value]) => `${key}: ${value}`).join('\n    ')}
+  }`).join(',  ') %>
+}
 
 export default async function NuxtStatorPlugin (ctx, inject) {
   const hydrate = (initialState) => {
     return process.client
-      ? Vue.observable(<%= options.isSPA ? 'initialState' : 'window.__NUXT__.$state' %>)
+      ? <%= options.isSPA ? 'initialState' : 'window.__NUXT__.$state' %>
       : initialState
   }
 
   const initialState = await state(ctx)
 
-  createStore({
-    ctx,
```

Comment:
Dont pass the Nuxt context to the store anymore. It shouldnt be needed and leads probably to possible bad design. Store creation should be fully independent of the the framework that is used.

If this is still needed, I believe we should do this the other way around: set the nuxt context on the store instead of the store on the nuxt context.